streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -73,203 +57,38 @@ public static boolean isValidCleanupPolicy(final String cleanupPolicy) {
         return true;
     }
 
-    private class ZKStringSerializer implements ZkSerializer {
-
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public byte[] serialize(Object data) {
-            try {
-                return ((String) data).getBytes("UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-
-        /**
-         * @throws AssertionError if the byte String encoding type is not supported
-         */
-        @Override
-        public Object deserialize(byte[] bytes) {
-            try {
-                if (bytes == null)
-                    return null;
-                else
-                    return new String(bytes, "UTF-8");
-            } catch (UnsupportedEncodingException e) {
-                throw new AssertionError(e);
-            }
-        }
-    }
-
     public InternalTopicManager() {
-        this.zkClient = null;
         this.replicationFactor = 0;
         this.windowChangeLogAdditionalRetention = WINDOW_CHANGE_LOG_ADDITIONAL_RETENTION_DEFAULT;
     }
 
-    public InternalTopicManager(String zkConnect, final int replicationFactor, long windowChangeLogAdditionalRetention) {
-        this.zkClient = new ZkClient(zkConnect, 30 * 1000, 30 * 1000, new ZKStringSerializer());
+    public InternalTopicManager(StreamsConfig config, final int replicationFactor, long windowChangeLogAdditionalRetention) {
```

Comment:
A better way to do this would be to pass in the StreamsKafkaClient, i.e.,
`public InternalTopicManager(final StreamsKafkaClient streamsKafkaClient, final int replicationFactor, final long windowChangeLogAdditionalRetention)`

Why? well the `StreamsConfig` is only used to construct the `StreamsKafkaClient` and it would mean in unit tests we can pass in a Stub or Mock  for `StreamsKafkaClient`. We can then remove the no-arg constructor on line 60 (It is only used in `MockInternalTopicManager`)